### PR TITLE
Shell sp19

### PR DESCRIPTION
**Pull request by otsneh**
Added new `ps` with `/proc` and `pfd` command.
Moved week 2 parts to week 2 section, and added grading section.

### CODE REVIEW

File: shell.md [hunk 1]
```diff
@@ -397,3 +358,95 @@ Input: `x; y`
...
+
+Like our good old `ps`, your shell should print out information about all currently executing processes. You should include the shell and its immediate children, but don't worry about grandchildren or other processes. Make sure you use `print_process_info_header()` and `print_process_info()` (and maybe some other helper functions)!
+
+_Note:_ while `ps` is normally a separate binary, it is a built-in command for your shell. (This is not "execing `ps`", this is you implementing it in the code. Thus you may have to keep track of some information for each process.)
+
+Your version of the `ps` should print the following information for each process:
+- PID: The pid of the process
+- NLWP: The number of threads currently being used in the process
+- VSZ: The program size (virtual memory size) of the process, in kilobytes
+- STAT: The state of the process
```
[the remark this answers is not in the document]
Comment (otsneh) [reply shell.md]: Not a typo, `/bin/ps` uses STAT so I kept the names the same. I can change to state if you want though.

File: shell.md [hunk 2]
```diff
@@ -397,3 +358,95 @@ Input: `x; y`
...
+- START: The start time of the process
+- TIME: The amount of cpu time that the process has been executed for
+- COMMAND: The command that executed the process
+
+*Hint:* You may find the `/proc` filesystem to be useful, as well as the man pages for it.
```
[the remark this answers is not in the document]
Comment (ASankaran) [reply shell.md]: https://en.wikipedia.org/wiki/Procfs

"The proc filesystem..."

File: shell.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (redsn0w422) [reply shell.md]: Makes sense, STAT sounds good to me then.

File: shell.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (otsneh) [reply shell.md]: I was debating between filesystem and directory. `/proc` is mentioned everywhere on the internet as the `/proc` filesystem so I figured I'd keep it standard. I can change it to directory (or something else) if you'd still like

[the remark this answers is not in the document]
Comment (redsn0w422) [reply shell.md]: TIL, maybe we should include a bit about what proc is at the bottom of the docs then.

**Pull request merged**